src/redisraft.c

```diff
@@ -401,101 +577,36 @@ static bool handleInterceptedCommands(RedisRaftCtx *rr,
         return true;
     }
 
-    return false;
-}
-
-/* When sharding is enabled, handle sharding aspects before processing
- * the request:
- *
- * 1. Compute hash slot of all associated keys and validate there's no cross-slot
- *    violation.
- * 2. Update the request's hash_slot for future reference.
- * 3. If the hash slot is associated with a foreign ShardGroup, perform a redirect.
- * 4. If the hash slot is not mapped, produce a CLUSTERDOWN error.
- */
-static RRStatus handleSharding(RedisRaftCtx *rr,
-                               RedisModuleCtx *ctx, RaftRedisCommandArray *cmds)
-{
-    int slot;
-
-    if (computeHashSlot(rr, cmds, &slot) != RR_OK) {
-        RedisModule_ReplyWithError(ctx, "CROSSSLOT Keys in request don't hash to the same slot");
-        return RR_ERROR;
-    }
-
-    /* If command has no keys, continue */
-    if (slot == -1) {
-        return RR_OK;
-    }
-
-    /* Make sure hash slot is mapped and handled locally. */
-    ShardGroup *sg = rr->sharding_info->stable_slots_map[slot];
-    if (!sg) {
-        RedisModule_ReplyWithError(ctx, "CLUSTERDOWN Hash slot is not served");
-        return RR_ERROR;
+    if (len == strlen("MIGRATE") && strncasecmp(cmd_str, "MIGRATE", len) == 0) {
+        handleMigrateCommand(rr, ctx, cmd);
+        return true;
     }
 
-    /* If accessing a foreign shardgroup, issue a redirect. We use round-robin
-     * to all nodes to compensate for the fact we do not have an up-to-date knowledge
-     * of who the leader is and whether or not some configuration has changed since
-     * last refresh (when refresh is implemented, in the future).
-     */
-    if (!sg->local) {
-        if (sg->next_redir >= sg->nodes_num) {
-            sg->next_redir = 0;
-        }
-        replyRedirect(ctx, slot, &sg->nodes[sg->next_redir++].addr);
-        return RR_ERROR;
+    if (len == strlen("ASKING") && strncasecmp(cmd_str, "ASKING", len) == 0) {
+        handleAsking(rr, ctx);
+        return true;
     }
 
-    return RR_OK;
+    return false;
 }
 
-static void handleRedisCommand(RedisRaftCtx *rr,
-                               RedisModuleCtx *ctx, RaftRedisCommandArray *cmds)
+static RRStatus handleRedisCommandAppend(RedisRaftCtx *rr,
```

Comment:
We are not using return value anywhere as far as I can see. Do we need this return code?

Reply:
good Q, investigating.  code doesn't quite make sense to me anymore.  rereading handleRedisCommandAppend to understand what we want